code/game/machinery/harvester.dm

```diff
@@ -159,6 +159,7 @@
 
 /obj/machinery/harvester/on_emag(mob/user)
 	..()
+	is_emagged = TRUE
```

Comment:
Theres a machine flag for determining if something is emagged. This flag is set in the parent call of on_emag, which means you don't need to store this.

Parent call:
```
/// Performs the actions to emag something, given that should_emag succeeded. You should NOT call this directly. Call use_emag.
/obj/proc/on_emag(mob/user)
	SHOULD_CALL_PARENT(TRUE)
	if(emag_toggleable)
		obj_flags ^= EMAGGED
	else
		obj_flags |= EMAGGED
```

In order to check if an object has been emagged, you can use the following check:
`if (obj_flags & EMAGGED)`
If you have multiple conditions in an if statement, then it is good practice to wrap your bitflags in brackets so that the order of operations doesn't get confused.
`if ((obj_flags & EMAGGED) && is_condition_met)`